File: Source/WebKit2/Shared/API/sling/SWKSecurityOrigin.cpp
/*
 * Copyright (C) 2014 Naver Corp. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *  * Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 *  * Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
 * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
 * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
 * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

#include "config.h"
#include "SWKSecurityOrigin.h"

namespace webkit {

void SWKSecurityOrigin::INIT(const std::string& protocol, const std::string& host, int32_t port)
{
    m_protocol = protocol;
    m_host = host;
    m_port = port;
    nativeCreate();
}

const std::string& SWKSecurityOrigin::getProtocol() const
{
    return m_protocol;
}

const std::string& SWKSecurityOrigin::getHost() const
{
    return m_host;
}

int32_t SWKSecurityOrigin::getPort() const
{
    return m_port;
}

} // namespace webkit

using namespace webkit;

namespace WebKit {
namespace Managed {

void SWKSecurityOrigin::INIT(const std::string&, const std::string&, int32_t)
{
}

}
} // namespace WebKit

File: Source/WebKit2/Shared/API/sling/SWKSecurityOrigin.java
package labs.naver.webkit;

import labs.naver.onig.CalledByNative;
import labs.naver.onig.NativeClassExport;
import labs.naver.onig.NativeConstructor;
import labs.naver.onig.NativeDestructor;
import labs.naver.onig.NativeNamespace;
import labs.naver.onig.NativeObjectField;

@NativeNamespace("WebKit")
@NativeClassExport("WEBKIT_EXPORT")
public class SWKSecurityOrigin {
    private final String mProtocol;
    private final String mHost;
    private final int mPort;

    @NativeObjectField
    private int mNativeClass;

    @CalledByNative
    private SWKSecurityOrigin(String protocol, String host, int port) {
        mProtocol = protocol;
        mHost = host;
        mPort = port;

        nativeCreate();
    }

    @Override
    protected void finalize() {
        nativeDestroy();
    }

    public String getProtocol() {
        return mProtocol;
    }

    public String getHost() {
        return mHost;
    }

    public int getPort() {
        return mPort;
    }

    @NativeConstructor
    private native void nativeCreate();
    @NativeDestructor
    private native void nativeDestroy();
}
